cli/cmd/agent_aws-install_ec2ssm.go

```diff
@@ -0,0 +1,350 @@
+
+This command will create a role and instance profile with 'SSMManagedInstanceCore'
+attached and associate that instance profile with the target instances. If the target
+instances already have associated instance profiles, this command will not change
+their state. This command will teardown the IAM role and instance profile before exiting.
+
+This command authenticates with AWS credentials from well-known locations on the user's
+machine. The principal associated with these credentials should have the
+'AmazonEC2FullAccess', 'IAMFullAccess' and 'AmazonSSMFullAccess' policies attached.
+
+Target instances must have the SSM agent installed and running for successful
+installation.
+
+To skip IAM role / instance profile creation and instance profile association:
+
+    lacework agent aws-install ec2ssm --skip_iam_role_creation
+
+To provide a preexisting IAM role with the 'SSMManagedInstanceCore' policy
+
+    lacework agent aws-install ec2ssm --iam_role_name IAMRoleName
+
+To filter by one or more regions:
+
+    lacework agent aws-install ec2ssm --include_regions us-west-2,us-east-2
+
+To filter by instance tag:
+
+    lacework agent aws-install ec2ssm --tag TagName,TagValue
+
+To filter by instance tag key:
+
+    lacework agent aws-install ec2ssm --tag_key TagName
+
+To provide an agent access token of your choice, use the command 'lacework agent token list',
+select a token and pass it to the '--token' flag. This flag must be selected if the
+'--noninteractive' flag is set.
+
+    lacework agent aws-install ec2ssm --token <token>
+
+AWS credentials are read from the following environment variables:
+- AWS_ACCESS_KEY_ID
+- AWS_SECRET_ACCESS_KEY
+- AWS_SESSION_TOKEN (optional)
+- AWS_REGION`,
+	}
+)
+
+func init() {
+	// 'agent aws-install ec2ssm' flags
+	agentInstallAWSSSMCmd.Flags().StringVar(&agentCmdState.InstallTagKey,
+		"tag_key", "", "only install agents on infra with this tag key set",
+	)
+	agentInstallAWSSSMCmd.Flags().StringSliceVar(&agentCmdState.InstallTag,
+		"tag", []string{}, "only install agents on infra with this tag",
+	)
+	agentInstallAWSSSMCmd.Flags().StringSliceVarP(&agentCmdState.InstallIncludeRegions,
+		"include_regions", "r", []string{}, "list of regions to filter on",
+	)
+	agentInstallAWSSSMCmd.Flags().StringVar(&agentCmdState.InstallAgentToken,
+		"token", "", "agent access token",
+	)
+	agentInstallAWSSSMCmd.Flags().IntVarP(
+		&agentCmdState.InstallMaxParallelism,
+		"max_parallelism",
+		"n",
+		50,
+		"maximum number of workers executing AWS API calls, set if rate limits are lower or higher than normal",
+	)
+	agentInstallAWSSSMCmd.Flags().StringVar(
+		&agentCmdState.InstallBYORole,
+		"iam_role_name",
+		"",
+		"IAM role name (not ARN) with SSM policy, if not provided then an ephemeral role will be created",
+	)
+	agentInstallAWSSSMCmd.Flags().BoolVar(
+		&agentCmdState.InstallSkipCreatInfra,
+		"skip_iam_role_creation",
```

Comment:
also very nitpicky, TIOLI: it's more common for unix commands to use kebab-case instead of snake_case for flags. Bash/zsh/etc don't distinguish between underscores and dashes, they're both just word chars.

Reply:
I agree, this is just following the existing flag style for the Lacework CLI